python/ray/data/preprocessors/concatenator.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Optional
+import numpy as np
+import pandas as pd
+
+from ray.data.extensions import TensorArray
+from ray.data.preprocessor import Preprocessor
+
+
+class Concatenator(Preprocessor):
+    """Create tensor columns via concatenation.
+
+    A tensor column is a a column consisting of ndarrays as elements.
+
+    Example:
+        >>> import pandas as pd
+        >>> from ray.data.preprocessors import Concatenator
+        >>> df = pd.DataFrame({"a": [1, 2, 3, 4], "b": [1, 2, 3, 4],})
+        >>> ds = ray.data.from_pandas(df)
+        >>> prep = Concatenator(["a", "b"], "c")
+        >>> new_ds = prep.transform(ds)
+        >>> df = new_ds.to_pandas()
+        #         c
+        # 0  [1, 1]
+        # 1  [2, 2]
+        #       ...
+        >>> x = df["c"].iloc[0]
+        >>> assert x.to_numpy().tolist() == [1, 1]
+
+    Args:
+        output_column: output_column is a string that represents the
+            name of the outputted, concatenated tensor.
+        exclude: A list of column names that should be excluded
+            from concatenation. All other columns will be dropped.
+        dtype: Optional. The dtype to convert the output column array to.
+
+    Raises:
+        ValueError if any element in `columns` does not exist in the dataset.
+    """
+
+    _is_fittable = False
+
+    def __init__(
+        self,
+        output_column: str,
+        exclude: Optional[List[str]] = None,
```

Comment:
You should also have an "include" field (exclude shouldn't be the only way to do it).

Reply:
added